ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -280,7 +281,12 @@ private Callback<String> routerStitchBlobCallback(BlobInfo blobInfo) {
     private Callback<String> idConverterCallback(BlobInfo blobInfo, String blobId) {
       return buildCallback(frontendMetrics.putIdConversionMetrics, convertedBlobId -> {
         restResponseChannel.setHeader(RestUtils.Headers.LOCATION, convertedBlobId);
-        if (blobInfo.getBlobProperties().getTimeToLiveInSeconds() == Utils.Infinite_Time) {
+        // TODO [S3] Probably we don't reuse NamedBlobPutHandler but write a new S3PutHandler.
+        if (S3MultipartUploadHandler.isUploadPartRequest(restRequest)) {
+          // For s3 part upload, we don't need to do TtlUpdate, named blob db update etc. It only needs the PutBlob
```

Comment:
QQ. I think by default the blobs are permanent if the request doesn't contain `x-ambry-ttl` header ([Reference code](https://github.com/linkedin/ambry/blob/24e12d786152d9f5e8f87a0707ef20a75102de86/ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java#L580)). Should we ourselves add `x-ambry-ttl` header in [S3MultipartUploadPartHandler](https://github.com/linkedin/ambry/blob/24e12d786152d9f5e8f87a0707ef20a75102de86/ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3MultipartUploadPartHandler.java#L64)?